### PR TITLE
Ensure unique ordering for function get_all_nodes

### PR DESCRIPTION
**Pull request by knarfnitram**
During #126 I encountered non-deterministic behavior for the function `get_all_nodes`, when they are used from a geometry based on a line set.

To ensure the same numbering of the nodes, some adjustments were made and a test case is added.

### CODE REVIEW

File: meshpy/geometry_set.py [hunk 1]
```diff
@@ -219,7 +219,8 @@ def get_all_nodes(self):
             nodes = []
             for element in self.geometry_objects[mpy.geo.line]:
                 nodes.extend(element.nodes)
-            return list(set(nodes))
+            # Remove duplicates while preserving order
+            return list(dict.fromkeys(nodes))
```
Comment (isteinbrecher) [inline meshpy/geometry_set.py:223]: This seems to resolve the issue without any additional manual tweaking.

File: tests/testing_meshpy.py [hunk 2]
```diff
@@ -541,6 +541,36 @@ def test_meshpy_geometry_sets(self):
         # Check the output.
         compare_test_result(self, mesh.get_string(header=False))
 
+    def test_unique_ordering_of_line_condition(self):
+        """
+        This test ensures that the ordering of the nodes returned from the function get_all_nodes is unique for line sets.
+        """
```
[the remark this answers is not in the document]
Comment (knarfnitram) [reply tests/testing_meshpy.py]: Well, should this be not consistent in meshpy? See function `test_meshpy_get_nodes_by_function`

[the remark this answers is not in the document]
Comment (isteinbrecher) [reply tests/testing_meshpy.py]: Yes, some docstrings are still "wrong" I never adapted all of them.

[the remark this answers is not in the document]
Comment (davidrudlstorfer) [reply tests/testing_meshpy.py]: This will be solved once a good pre-commit hook is in place and also checks/formats docstrings

**Comment by isteinbrecher** [comment]
@knarfnitram thanks for the changes! Performance tests also pass, so the added sort functionality should at least not completely break the performance. Feel free to merge.

**Comment by knarfnitram** [comment]
Thanks @isteinbrecher for helping with the sort. 
This MR is still blocked due to a missing approval ;)

**Pull request merged**